hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/OmSnapshotLocalDataManager.java

```diff
@@ -75,6 +93,7 @@ public void computeAndSetChecksum(Yaml yaml, OmSnapshotLocalData data) throws IO
       }
     };
     this.versionNodeMap = new HashMap<>();
```

Comment:
Is `ConcurrentHashMap` better?

Reply:
We need not use ConcurrentHashMap since graph is not concurrent and can be only updated inside a synchrnized block. If you see both the Map and graph data structure only gets updated in the upsertNode function which is synchronized

Reply:
We can read in parallel when there is no write lock involved

Reply:
locking is good but lock alone may not be sufficient.

see Java memory model. related: https://stackoverflow.com/a/15794355

Reply:
Yeah I have synchornized reads and writes on concurrentHashMap and localDataGraph using an internalReadWriteLock